compiler/rustc_hir_typeck/src/pat.rs

```diff
@@ -1257,7 +1256,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                 subpats,
                 &variant.fields.raw,
                 expected,
-                had_err,
+                had_err.is_some(),
```

Comment:
unless you want to backport this, can you change `emit_err_pat_wrong_number_of_fields` to also take `Result<(), ErrorGuaranteed>`.

also, can you change methods which return `Option<Diag<'tcx>>` to `Result<(), Diag<'tcx>>` unless these are methods which try to build an error while already on the failure path.